### PR TITLE
Fixing signedness conversion warnings in libevmasm

### PR DESCRIPTION
**Pull request by mijovic**
Part of #9082

### CODE REVIEW

**Comment by hrkrshnn** [comment]
```diff
diff -u /tmp/new.json /tmp/old.json
--- /tmp/new.json	2020-06-03 11:53:45.142968792 +0530
+++ /tmp/old.json	2020-06-03 11:53:25.744893010 +0530
@@ -15,7 +15,7 @@
             "linkReferences": null,
             "object": "bytecode removed",
             "opcodes": "opcodes removed",
-            "sourceMap": "36:96:0:-:0;;;;5:9:18446744073709551615;2:2;;;27:1;24;17:12;2:2;36:96:0;;;;;;;;;;;;;;;;12:1:18446744073709551615;9;2:12;74:56:0;;;:::i;:::-;;;;;;;;;;;;;;;;;;;;108:7;126:1;119:8;;74:56;:::o"
+            "sourceMap": "36:96:0:-:0;;;;5:9:-1;2:2;;;27:1;24;17:12;2:2;36:96:0;;;;;;;;;;;;;;;;12:1:-1;9;2:12;74:56:0;;;:::i;:::-;;;;;;;;;;;;;;;;;;;;108:7;126:1;119:8;;74:56;:::o"
           }
         }
       }
```

For reference, this is where CI fails. Looks like one of the output must be `-1`.

File: libevmasm/AssemblyItem.cpp [hunk 1]
```diff
@@ -316,8 +316,8 @@ std::string AssemblyItem::computeSourceMapping(
 
 	int prevStart = -1;
 	int prevLength = -1;
-	int prevSourceIndex = -1;
-	size_t prevModifierDepth = -1;
+	size_t prevSourceIndex = numeric_limits<size_t>::max();
+	size_t prevModifierDepth = numeric_limits<size_t>::max();
```
Comment (hrkrshnn) [inline libevmasm/AssemblyItem.cpp]:
```suggestion
	int prevSourceIndex = -1;
	int prevModifierDepth = -1;
```

File: libevmasm/AssemblyItem.cpp [hunk 2]
```diff
@@ -326,10 +326,10 @@ std::string AssemblyItem::computeSourceMapping(
 
 		SourceLocation const& location = item.location();
 		int length = location.start != -1 && location.end != -1 ? location.end - location.start : -1;
-		int sourceIndex =
+		size_t sourceIndex =
```
Comment (hrkrshnn) [inline libevmasm/AssemblyItem.cpp]:
```suggestion
		int sourceIndex =
```

File: libevmasm/AssemblyItem.cpp [hunk 3]
```diff
@@ -326,10 +326,10 @@ std::string AssemblyItem::computeSourceMapping(
...
 			location.source && _sourceIndicesMap.count(location.source->name()) ?
 			_sourceIndicesMap.at(location.source->name()) :
```
Comment (hrkrshnn) [inline libevmasm/AssemblyItem.cpp]:
```suggestion
			static_cast<int>(_sourceIndicesMap.at(location.source->name())) :
```

File: libevmasm/AssemblyItem.cpp [hunk 4]
```diff
@@ -326,10 +326,10 @@ std::string AssemblyItem::computeSourceMapping(
...
-			-1;
+			numeric_limits<size_t>::max();
```
Comment (hrkrshnn) [inline libevmasm/AssemblyItem.cpp]:
```suggestion
			-1;
```

**Review by chriseth — APPROVED** [review]

**Pull request merged**